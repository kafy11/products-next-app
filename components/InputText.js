import { useRef } from 'react';

const InputText = ({ label, append, prepend, onChange, value = '', ...rest }) => {
    const inputRef = useRef(null);

    const handleLabelClick = () => inputRef.current.focus();
    const handleChange = (e) => onChange && onChange(e.target.value);

    const renderInputGroupPart = (part, content) => (
        <div className={`input-group-${part}`}>
            <span className="input-group-text">{content}</span>
        </div>
    );

    const inputField = (
        <input 
            type="text" 
            className="form-control" 
            ref={inputRef}
            onChange={handleChange}
            value={value}
            {...rest}
        />
    );

    if(!append && !prepend){
        return (
            <div className="form-group">
                <label onClick={handleLabelClick}>{label}</label>
                {inputField}
            </div>      
        );
    }
    
    return (
        <div className="form-group">
            <label onClick={handleLabelClick}>{label}</label>
            <div className="input-group mb-3">
                {renderInputGroupPart('prepend', prepend)}
                {inputField}
                {renderInputGroupPart('append', append)}
            </div>
        </div>
    )
}

export default InputText;