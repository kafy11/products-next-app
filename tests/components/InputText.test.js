import { shallow } from 'enzyme';
import { expectToExist } from '../testHelpers';
import InputText from '../../components/InputText';

let handleChange, wrapper;
const props = {
    label: "PreÃ§o",
    prepend: "$",
    append: ".00",
    value : "30",
    testProp: 2,
};

beforeEach(() => {
    handleChange = jest.fn();
    wrapper = shallow(
        <InputText 
            {...props}
            onChange={handleChange}
        />
    );
});

describe('InputText', () => {
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should handle onChange', () => {
        const newValue = "40";
        wrapper.find('input').simulate('change', { target: { value: newValue } });
        expect(handleChange).toHaveBeenLastCalledWith(newValue);
    });
    
    it('should render the label correctly', () => {
        const labelText = wrapper.find('label').text();
        expect(labelText).toBe(props.label);
    });
    
    it('should render the append correctly', () => {
        expectToExist(wrapper, '.input-group-append');
    });
    
    it('should render the prepend correctly', () => {
        expectToExist(wrapper, '.input-group-prepend');
    });
    
    it('should render the value correctly', () => {
        const value = wrapper.find('input').prop('value');
        expect(value).toBe(props.value);
    });
    
    it('should add the extra props to the input', () => {
        const testProp = wrapper.find('input').prop('testProp');
        expect(testProp).toBe(props.testProp);
    });
});