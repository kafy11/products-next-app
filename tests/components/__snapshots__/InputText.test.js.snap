// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText should render correctly 1`] = `
<div
  className="form-group"
>
  <label
    onClick={[Function]}
  >
    Pre√ßo
  </label>
  <div
    className="input-group mb-3"
  >
    <div
      className="input-group-prepend"
    >
      <span
        className="input-group-text"
      >
        $
      </span>
    </div>
    <input
      className="form-control"
      onChange={[Function]}
      testProp={2}
      type="text"
      value="30"
    />
    <div
      className="input-group-append"
    >
      <span
        className="input-group-text"
      >
        .00
      </span>
    </div>
  </div>
</div>
`;
